startup.py

```diff
@@ -0,0 +1,136 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import sys
+import re
+import glob
+
+from sgtk.platform import SoftwareLauncher, SoftwareVersion, LaunchInformation
+
+class PhotoshopLauncher(SoftwareLauncher):
+    """
+    Handles the launching of Photoshop. Contains the logic for
+    scanning for installed versions of the software and
+    how to correctly set up a launch environment for the tk-photoshopcc
+    engine.
+    """
+
+    def __init__(self, *args, **kwargs):
+        """
+        Constructor
+        """
+        # init base class
+        super(PhotoshopLauncher, self).__init__(*args, **kwargs)
+        # define our minimum supported version
+        self.minimum_supported_version = "2015.5"
```

Comment:
I'm thinking about other places in engine code where we check for the minimum supported version, typically in `init_engine`: 

- https://github.com/shotgunsoftware/tk-houdini/blob/master/engine.py#L41-L42
- https://github.com/shotgunsoftware/tk-maya/blob/master/engine.py#L262-L268
- https://github.com/shotgunsoftware/tk-nuke/blob/master/engine.py#L158-L164

Maybe a setting isn't the right place, but just thinking the value should be centralized/standardized for our engines now that we're using it in the sw launchers? Maybe a separate ticket if you think it's worth addressing?